### DISCUSSION TITLE
Question about "Last Session" value

### DISCUSSION THREAD

**Discussion by AdmiralRaccoon** [netalertx/NetAlertX#701] [General]
Hi, just a question of how the "Last Session" value is meant to be understand. For example, I have 2 Proxmox Hosts, PVE01 and PVE02. 
PVE01 ist up once a week for and hour to do backups, PVE02 runs 24/7

Why does :
PVE01 has a Last Session timestamp from June 3rd, even if it was last online on June 10th?
PVE02 has a Last Session Timestamp from March?

![image](https://github.com/jokob-sk/NetAlertX/assets/41727850/db2649a8-5c13-40fa-88c8-b9da7e4ac1f4)

Thanks for clearing up my confusion,
Philipp

**Comment by jokob-sk** [comment]
Hi @AdmiralRaccoon ,

I haven't touched this functionality since I started on this project so I might be wrong, but this is the SQL query responsible for updating the Last session:

```python
    sql.execute(f"""UPDATE Devices SET dev_LastConnection = '{startTime}',
                        dev_PresentLastScan = 1
                    WHERE dev_PresentLastScan = 0
                      AND EXISTS (SELECT 1 FROM CurrentScan 
                                  WHERE dev_MAC = cur_MAC) """)
```
From the looks of it, it's the last time the device went offline. 

And `dev_FirstConnection` is updated the first time the device is detected, so the first time when it is connected to the network.

Let me know if you see it the same way or if I'm missing something. If you think my explanation is correct I might update the labels/descriptions to something more understandable.

**Comment by AdmiralRaccoon** [comment]
"From the looks of it, it's the last time the device went offline." - 
Yes, I would see that the same way. And this is what makes me wonder why, in my example, the PVE01 Last Session time is 2024-06-03, although it was online for a couple of hours on 2024-06-10.

Could it be a scanning script that went into a timeout, so it couldn´t get updated the tables during a scan maybe?

**Reply by jokob-sk** [reply]
Yeah, that could have happened. Or if e.g. a container restarted during a scan cycle. 

Based on the above I'll probably rename the label on that column, to something e.g. "Last Offline" - what do you think?
